### PR TITLE
Features/tensor refactor

### PR DESCRIPTION
**Pull request by bassoy**
1. Renamed and restructured extents types
2. Removed type traits for extents and strides types
3. Removed the old strides type. Strides are now the base type of the corresponding extents type (resolving issue #111 )
4. Created common non-member functions for extents
5. Created special non-member for constexpr evaluation of static extents traits
6. Created new Matlab like functions such as `ones` and `zeros`
7. Removed old CI files for travis and appveyor
8. Removed unnecessary CI badges in README

### CODE REVIEW

**Review by amitsingh19975 — COMMENTED** [review]
One step closer to improve the uBLAS.

File: test/tensor/test_static_tensor_matrix_vector.cpp [hunk 1]
```diff
@@ -442,17 +439,13 @@ BOOST_FIXTURE_TEST_CASE_TEMPLATE( test_tensor_matrix_vector_expressions, value,
         BOOST_CHECK_EQUAL (  T.rank() , Q.rank() );
         BOOST_CHECK       ( !T.empty()    );
 
-        for(auto i = 0ul; i < T.size(); ++i){
+        for(auto i = size_t{0}; i < T.size(); ++i){
             auto n = e[1];
-            auto ab = n * (n+1) / 2;
+            auto ab = value(n*(n+1)/2);
```
[the remark this answers is not in the document]
Comment (bassoy) [reply test/tensor/test_static_tensor_matrix_vector.cpp:444]: Is fixed now using `std::div`

File: test/tensor/test_tensor_matrix_vector.cpp [hunk 2]
```diff
@@ -442,7 +443,7 @@ BOOST_FIXTURE_TEST_CASE_TEMPLATE( test_tensor_matrix_vector_expressions, value,
 
         for(auto i = 0ul; i < T.size(); ++i){
             auto n = e[1];
-            auto ab = n * (n+1) / 2;
+            auto ab = value(n * (n+1) / 2);
```
[the remark this answers is not in the document]
Comment (bassoy) [reply test/tensor/test_tensor_matrix_vector.cpp:446]: Is fixed now using `std::div`

File: include/boost/numeric/ublas/tensor/extents/extents_dynamic_size.hpp [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+
+
+/** @brief Template class for storing tensor extents with runtime variable size.
+ *
+ * Proxy template class of std::vector<int_type>.
+ *
+ */
+template<integral T>
+class extents_core<T>
```
[the remark this answers is not in the document]
Comment (bassoy) [reply include/boost/numeric/ublas/tensor/extents/extents_dynamic_size.hpp:39]: Implemented the copy-swap idiom for the [assignment operator](https://en.cppreference.com/w/cpp/language/copy_assignment).

File: include/boost/numeric/ublas/tensor/tensor/tensor_dynamic.hpp [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+#include "tensor_engine.hpp"
+
+
+namespace boost::numeric::ublas {
+
+template<class V, class L>
+using engine_tensor_dynamic = tensor_engine<extents<>, L, std::vector<V>>;
+
+template<class V, class L>
+  class tensor_core<engine_tensor_dynamic<V,L>>
```
[the remark this answers is not in the document]
Comment (bassoy) [reply include/boost/numeric/ublas/tensor/tensor/tensor_dynamic.hpp:40]: Implemented the copy-swap idiom for the [assignment operator](https://en.cppreference.com/w/cpp/language/copy_assignment) using pass by value. Clang-Tidy does not see it.

File: include/boost/numeric/ublas/tensor/tensor/tensor_static.hpp [hunk 5]
```diff
@@ -0,0 +1,456 @@
...
+namespace boost::numeric::ublas::detail
+{
+template<class V, class L, std::size_t ... ns>
+using engine_tensor_static = tensor_engine<
+  extents<ns...>, L, std::array<V,product_v<extents<ns...>>> >;
+} // namespace boost::numeric::ublas::detail
+
+namespace boost::numeric::ublas {
+template<class V, class L, std::size_t ... ns>
+class tensor_core<detail::engine_tensor_static<V,L,ns...>>
```
[the remark this answers is not in the document]
Comment (bassoy) [reply include/boost/numeric/ublas/tensor/tensor/tensor_static.hpp:45]: Implemented the copy-swap idiom for the [assignment operator](https://en.cppreference.com/w/cpp/language/copy_assignment) using pass by value. Clang-Tidy does not see it.

File: include/boost/numeric/ublas/tensor/tensor/tensor_static_rank.hpp [hunk 6]
```diff
@@ -0,0 +1,473 @@
...
+
+
+
+namespace boost::numeric::ublas {
+
+template<class V, class L, std::size_t N>
+using engine_tensor_static_rank = tensor_engine<extents<N>, L, std::vector<V>>;
+
+template<class V, class L, std::size_t N>
+  class tensor_core<engine_tensor_static_rank<V,L,N>>
```
[the remark this answers is not in the document]
Comment (bassoy) [reply include/boost/numeric/ublas/tensor/tensor/tensor_static_rank.hpp:41]: Implemented the copy-swap idiom for the [assignment operator](https://en.cppreference.com/w/cpp/language/copy_assignment) using pass by value. Clang-Tidy does not see it.

**Comment by bassoy** [comment]
> improve

**Pull request merged**